### PR TITLE
Use function components with hooks

### PR DESCRIPTION
**Pull request by kfarnung**
Class-based components are being replaced with function-based components
that use hooks. Convert a bunch of the simple components and defer the
rest until later on.

### CODE REVIEW

File: src/Resistor/ResistorInput.tsx [hunk 1]
```diff
@@ -1,196 +1,105 @@
...
-  private colorByName(name: String) {
-    for (let color of Resistor.colorTable) {
-      if (name === color.name) {
-        return color;
-      }
-    }
+function ResistorInput() {
+  const initialBands = [Resistor.BLACK, Resistor.BLACK, Resistor.BLACK, null, Resistor.BROWN];
+  const [value, setValue] = useState(getResistorValue(initialBands));
+  const [bands, setBands] = useState(initialBands);
```
Comment (kfarnung) [inline src/Resistor/ResistorInput.tsx]: This is the extent of the hooks usage so far.

**Comment by kfarnung** [comment]
Most of this is just moving code around, but I did refactor the Resistor code pretty significantly.

File: src/Resistor/ResistorColorSelector.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+interface ColorSelectorProps {
+  colors: Color[];
+  index: number;
+  title: string;
+  onChange?: (index: number, color?: Color) => void;
+}
+
+function colorByName(name: String) {
+  for (let color of Resistor.colorTable) {
```
Comment (beckbria) [inline src/Resistor/ResistorColorSelector.tsx]: for (const color of Resistor.colorTable)

File: src/Resistor/ResistorInput.tsx [hunk 3]
```diff
@@ -1,196 +1,105 @@
...
-      return (
-        <use
-          xlinkHref="#Band"
-          x={this.getBandOffset(index)}
-          fill={band.colorCode}
-        />
-      );
+function getResistorValue(bands: (Color | null)[]) {
+  let colors: Color[] = [];
+  for (let i = 0; i < (bands.length - 1); ++i) {
```
Comment (beckbria) [inline src/Resistor/ResistorInput.tsx]: Add a comment explaining why we ignore the last band

**Pull request merged**